internal/brokers/broker_test.go

```diff
@@ -89,15 +111,29 @@ func TestGetAuthenticationModes(t *testing.T) {
 
 			b, _ := newBrokerForTests(t)
 
-			gotModes, err := b.GetAuthenticationModes(context.Background(), tc.sessionID, nil)
+			if tc.supportedUILayouts == nil {
+				tc.supportedUILayouts = []string{"required-value"}
+			}
+
+			// This is normally done in the broker's GetAuthenticationModes method, but we need to do it here to test the SelectAuthenticationMode method.
```

Comment:
you meant in the PAM module?

Reply:
What do you mean? I meant the validator's generation. They are generated in the GAM method, but since here we don't call it, we need to expose the function that generates the validators and call it manually.